rust/repositories.bzl

```diff
@@ -42,98 +55,272 @@ filegroup(
         target_triple = target_triple,
     )
 
-def BUILD_for_toolchain(name, target_triple):
-    """Emits a toolchain declaration for an existing toolchain workspace."""
+def BUILD_for_toolchain(workspace_name, name, exec_triple, target_triple):
+    """Emits a toolchain declaration to match an existing compiler and stdlib.
+
+    Args:
+      workspace_name: The name of the workspace that this toolchain resides in
+      name: The name of the toolchain declaration
+      exec_triple: The rust-style target that this compiler runs on
+      target_triple: The rust-style target triple of the tool
+    """
 
     system = triple_to_system(target_triple)
-    constraint_set = triple_to_constraint_set(target_triple)
 
-    constraint_set_strs = []
-    for constraint in constraint_set:
-        constraint_set_strs.append("\"{}\"".format(constraint))
+    exec_constraint_set = triple_to_constraint_set(exec_triple)
+    exec_constraint_set_strs = []
+    for constraint in exec_constraint_set:
+        exec_constraint_set_strs.append("\"{}\"".format(constraint))
+    exec_constraint_sets_serialized = "[{}]".format(", ".join(exec_constraint_set_strs))
 
-    constraint_sets_serialized = "[{}]".format(", ".join(constraint_set_strs))
+    target_constraint_set = triple_to_constraint_set(target_triple)
+    target_constraint_set_strs = []
+    for constraint in target_constraint_set:
+        target_constraint_set_strs.append("\"{}\"".format(constraint))
+    target_constraint_sets_serialized = "[{}]".format(", ".join(target_constraint_set_strs))
 
-    return """toolchain(
+    return """
+toolchain(
```

Comment:
toolchain needs to be separate from rust_toolchain to avoid the eager download

Reply:
This is now done (via a sibling repository rule)

I think this adds a pretty large amount of cognitive overhead for the usage, but it serves a purpose.